src/main/java/studio/magemonkey/fabled/api/player/PlayerData.java

```diff
@@ -1949,15 +1949,38 @@ public void updatePlayerStat(Player player) {
+        // Stats added in 1.21+
+        this.updateMCAttribute(player, Attribute.GENERIC_ATTACK_DAMAGE, AttributeManager.ATTACK_DAMAGE, 0.0f, 2048.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_ATTACK_KNOCKBACK, AttributeManager.ATTACK_KNOCKBACK, 0.0f, 5.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_FLYING_SPEED, AttributeManager.FLYING_SPEED, 0.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_FOLLOW_RANGE, AttributeManager.FOLLOW_RANGE, 0.0f, 2048.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_MAX_ABSORPTION, AttributeManager.ABSORPTION, 0.0f, 2048.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_SCALE, AttributeManager.SCALE, 0.0625f, 16.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_STEP_HEIGHT, AttributeManager.STEP_HEIGHT, 0.0f, 10.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_JUMP_STRENGTH, AttributeManager.JUMP_STRENGTH, 0.0f, 32.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_GRAVITY, AttributeManager.GRAVITY, -1.0f, 1.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_SAFE_FALL_DISTANCE, AttributeManager.SAFE_FALL_DISTANCE, -1024.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_FALL_DAMAGE_MULTIPLIER, AttributeManager.FALL_DAMAGE_MULTIPLIER, 0.0f, 100.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_BURNING_TIME, AttributeManager.BURNING_TIME, 0.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_EXPLOSION_KNOCKBACK_RESISTANCE, AttributeManager.EXPLOSION_KNOCKBACK_RESISTANCE, 0.0f, 1.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_MOVEMENT_EFFICIENCY, AttributeManager.MOVEMENT_EFFICIENCY, 0.0f, 1.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_OXYGEN_BONUS, AttributeManager.OXYGEN_BONUS, 0.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.GENERIC_WATER_MOVEMENT_EFFICIENCY, AttributeManager.WATER_MOVEMENT_EFFICIENCY, 0.0f, 1.0f);
+        // Player only stats 1.21+
+        this.updateMCAttribute(player, Attribute.PLAYER_BLOCK_INTERACTION_RANGE, AttributeManager.BLOCK_INTERACTION_RANGE, 0.0f, 64.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_ENTITY_INTERACTION_RANGE, AttributeManager.ENTITY_INTERACTION_RANGE, 0.0f, 64.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_BLOCK_BREAK_SPEED, AttributeManager.BLOCK_BREAK_SPEED, 0.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_MINING_EFFICIENCY, AttributeManager.MINING_EFFICIENCY, 0.0f, 1024.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_SNEAKING_SPEED, AttributeManager.SNEAKING_SPEED, 0.0f, 1.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_SUBMERGED_MINING_SPEED, AttributeManager.SUBMERGED_MINING_SPEED, 0.0f, 20.0f);
+        this.updateMCAttribute(player, Attribute.PLAYER_SWEEPING_DAMAGE_RATIO, AttributeManager.SWEEPING_DAMAGE_RATIO, 0.0f, 1.0f);
```

Comment:
We should probably wrap these update calls in a version check to ensure that these attributes are applicable to the version that is being used. Writing from the top of my head, but `if (Version.CURRENT.isAtLeast(Version.V1_21_R1))` should do it.

Reply:
Sounds good to me, I'll get to it sometime tomorrow.

Reply:
Added in version check!